rotkehlchen/exchanges/bitstamp.py

```diff
@@ -0,0 +1,711 @@
+            self,
+            response: Response,
+            case: Literal['trades'],
+    ) -> List[Trade]:
+        ...
+
+    @overload  # noqa: F811
+    def _process_unsuccessful_response(  # pylint: disable=no-self-use
+            self,
+            response: Response,
+            case: Literal['asset_movements'],
+    ) -> List[AssetMovement]:
+        ...
+
+    def _process_unsuccessful_response(
+            self,
+            response: Response,
+            case: str,
+    ) -> Union[
+        List[Trade],
+        List[AssetMovement],
+        Tuple[bool, str],
+        Tuple[Optional[Dict[Asset, Balance]], str],
+    ]:
+        """This function processes not successful responses for the following
+        cases listed in `case`.
+        """
+        case_pretty = case.replace('_', ' ')  # human readable case
+        # Process error response as a dict
+        try:
+            response_dict = rlk_jsonloads_dict(response.text)
+        except JSONDecodeError:
+            msg = f'Bitstamp returned invalid JSON response: {response.text}.'
+            log.error(msg)
+
+            # Custom response per case
+            if case in {'validate_api_key', 'balances'}:
+                raise RemoteError(msg)
+            elif case in {'trades', 'asset_movements'}:
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying Bistamp {case_pretty}: {msg}',
+                )
+                no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
+                return no_results
+            else:
+                raise AssertionError(f'Unexpected Bitstamp response_case: {case}.')
+
+        # Get error code
+        error_code = response_dict.get('code', None)
+
+        # Local system clock not synch
+        self._check_for_system_clock_not_synced_error(error_code)
+
+        # Errors related with the API key return a human redable message
+        if (
+            case == 'validate_api_key' and
+            error_code in set(API_KEY_ERROR_CODE_ACTION.keys())
+        ):
+            return False, API_KEY_ERROR_CODE_ACTION[response_dict['code']]
+
+        # Below any other error not related with the system clock or the API key
+        reason = response_dict.get('reason', None) or response.text
+        msg = (
+            f'Bitstamp query responded with error status code: {response.status_code} '
+            f'and text: {reason}.'
+        )
+        log.error(msg)
+
+        # Custom response per case
+        if case == 'validate_api_key':
+            raise RemoteError(msg)
+        if case == 'balances':
+            return None, msg
+        elif case in {'trades', 'asset_movements'}:
+            self.msg_aggregator.add_error(
+                f'Got remote error while querying Bistamp {case_pretty}: {msg}',
+            )
+            no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
```

Comment:
Both here and above where you do the same thing there is no need tospecify this type. You don't ever return any trades or asset movements from here. The error is correct. Try this:

```diff
diff --git a/rotkehlchen/exchanges/bitstamp.py b/rotkehlchen/exchanges/bitstamp.py
index f055312a..d3d16349 100644
--- a/rotkehlchen/exchanges/bitstamp.py
+++ b/rotkehlchen/exchanges/bitstamp.py
@@ -647,8 +647,7 @@ class Bitstamp(ExchangeInterface):
             response: Response,
             case: str,
     ) -> Union[
-        List[Trade],
-        List[AssetMovement],
+        List,
         Tuple[bool, str],
         Tuple[Optional[Dict[Asset, Balance]], str],
     ]:
@@ -670,8 +669,7 @@ class Bitstamp(ExchangeInterface):
                 self.msg_aggregator.add_error(
                     f'Got remote error while querying Bistamp {case_pretty}: {msg}',
                 )
-                no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
-                return no_results
+                return []
             else:
                 raise AssertionError(f'Unexpected Bitstamp response_case: {case}.')
 
@@ -705,7 +703,6 @@ class Bitstamp(ExchangeInterface):
             self.msg_aggregator.add_error(
                 f'Got remote error while querying Bistamp {case_pretty}: {msg}',
             )
-            no_results: Union[List[Trade], List[AssetMovement]] = []  # type: ignore
-            return no_results
+            return []
         else:
             raise AssertionError(f'Unexpected Bitstamp response_case: {case}.')
```

Reply:
good one, I tried many things not a single one the right one.